src/modules/src/error_estimator_ukf.c

```diff
@@ -0,0 +1,1623 @@
+            {
+              Pxy[jj] = 0.0f;
+            }
+
+            // loop over all sigma points
+            for (jj = 0; jj < (DIM_FILTER + 2); jj++)
+            {
+              for (ii = 0; ii < DIM_FILTER; ii++)
+              {
+                tmpSigmaVec[ii] = sigmaPoints[ii][jj];
+              }
+              computeOutputFlow_y(&outTmp, &tmpSigmaVec[0], &m.data.flow, gyroAverage);
+              Pyy = Pyy + weights[jj] * (outTmp - observation) * (outTmp - observation);
+
+              for (kk = 0; kk < DIM_FILTER; kk++)
+              {
+                Pxy[kk] = Pxy[kk] + weights[jj] * (tmpSigmaVec[kk] - xEst[kk]) * (outTmp - observation);
+              }
+            }
+            // Add TOF Noise R
+            Pyy = Pyy + (m.data.flow.stdDevY) * (m.data.flow.stdDevY);
+
+            innovation = m.data.flow.dpixely - observation;
+            meas_NY = m.data.flow.dpixely;
+            pred_NY = observation;
+
+            ukfUpdate(&Pxy[0], &Pyy, innovation);
+          }
+          break;
+        case MeasurementTypeBarometer:
+          //_________________________________________________________________________________
+          // UKF update - Baro
+          //_________________________________________________________________________________
+          // compute mean tdoa observation
+          if (initializedNav)
+          {
+            observation = 0.0f;
+            for (jj = 0; jj < (DIM_FILTER + 2); jj++)
+            {
+              for (ii = 0; ii < DIM_FILTER; ii++)
+              {
+                tmpSigmaVec[ii] = sigmaPoints[ii][jj];
+              }
+              computeOutputBaro(&outTmp, &tmpSigmaVec[0]);
+              observation = observation + weights[jj] * outTmp;
+            }
+
+            // initialize measurement and cross covariance
+            Pyy = 0.0f;
+            for (jj = 0; jj < DIM_FILTER; jj++)
+            {
+              Pxy[jj] = 0.0f;
+            }
+
+            // loop over all sigma points
+            for (jj = 0; jj < (DIM_FILTER + 2); jj++)
+            {
+              for (ii = 0; ii < DIM_FILTER; ii++)
+              {
+                tmpSigmaVec[ii] = sigmaPoints[ii][jj];
+              }
+              computeOutputBaro(&outTmp, &tmpSigmaVec[0]);
+              Pyy = Pyy + weights[jj] * (outTmp - observation) * (outTmp - observation);
+
+              for (kk = 0; kk < DIM_FILTER; kk++)
+              {
+                Pxy[kk] = Pxy[kk] + weights[jj] * (tmpSigmaVec[kk] - xEst[kk]) * (outTmp - observation);
+              }
+            }
+
+            // Add Baronoise R
+            Pyy = Pyy + measNoiseBaro;
+
+            innovation = (m.data.barometer.baro.asl-baroAslBias) - observation;
+            innoCheck = innovation * innovation / Pyy;
+
+            if (innoCheck < qualGateBaro)
+            {
+              ukfUpdate(&Pxy[0], &Pyy, innovation);
+            }
+          }
+          break;
+
+        case MeasurementTypeSweepAngle:
```

Comment:
I'm wondering how this could have been combined with the measurements model separated in the kalman estimator: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/kalman_core/mm_sweep_angles.c